### PR TITLE
Admin Fixes

### PR DESCRIPTION
**Pull request by chrisrodz**
Updating the Sequelize version broke some of the code (because Sequelize isn't very good). This PR fixes it. @MrMaksimize Please review and merge

### CODE REVIEW

**Comment by MrMaksimize** [comment]
Lol sequelize hater. Thanks for the fixes :) will review tonight or early
am :) also didn't see the pr but I'm hoping there's test coverage for the
things that broke :) I love u C!

On Thursday, July 17, 2014, Christian A. Rodriguez notifications@github.com
wrote:

> Updating the Sequelize version broke some of the code (because Sequelize
> isn't very good). This PR fixes it. @MrMaksimize
> 
> ## https://github.com/MrMaksimize Please review and merge
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/CoquiCoders/bizwallet admin-fixes
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/CoquiCoders/bizwallet/pull/43
> Commit Summary
> - URGENT ADMIN FIXES
> 
> File Changes
> - _M_ controllers/admin.js
>   https://github.com/CoquiCoders/bizwallet/pull/43/files#diff-0 (5)
> - _M_ views/admin/entry.jade
>   https://github.com/CoquiCoders/bizwallet/pull/43/files#diff-1 (2)
> - _M_ views/admin/list.jade
>   https://github.com/CoquiCoders/bizwallet/pull/43/files#diff-2 (8)
> 
> Patch Links:
> - https://github.com/CoquiCoders/bizwallet/pull/43.patch
> - https://github.com/CoquiCoders/bizwallet/pull/43.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/43.

## 

Sorry for the short message, this was sent from my mobile device.

File: views/admin/list.jade [hunk 1]
```diff
@@ -25,7 +25,7 @@ block content
               a.btn.btn-default.admin-edit(href="#{base}#{path}/#{entry.id}/edit") edit
               a.btn.btn-default.delete-model.admin-edit(href="#{base}#{path}/#{entry.id}/delete") delete
             td
-              a(href="#{base}#{path}/#{entry.id}") #{entry.selectedValues[fields.title.name]}
-            td= entry.selectedValues[fields.applicationDeadline.name]
-            td= entry.selectedValues[fields.benefitType.name]
-            td= entry.selectedValues[fields.agencyContactName.name]
+              a(href="#{base}#{path}/#{entry.id}") #{entry.dataValues[fields.title.name]}
```
Comment (MrMaksimize) [inline views/admin/list.jade]: We shouldn't be hardcoding values like these into the list template.  List template will be reusable.

The model should provide this information and the list should know how to get it.

There is already a base implementation of this in model utils

File: views/mixins/forms.jade [hunk 2]
```diff
@@ -22,7 +22,10 @@ mixin choiceSelectField(formOptions)
     select(class="form-control" name=formOptions.name multiple=(formOptions.widget == "multiSelect"))
       each choice, cVal in formOptions.choices
         // TODO look at exploding classes
-        option(value=cVal selected=(formOptions.value == cVal)) #{choice}
+        if Object.prototype.toString.call(formOptions.value) === '[object Array]'
```
Comment (MrMaksimize) [inline views/mixins/forms.jade]: Whoa -- that's an intense way of checking for an array. 

You could use an underscore method here -- or just typeof

File: app.js [hunk 3]
```diff
@@ -91,6 +91,7 @@ app.use(function(req, res, next) {
   res.locals.user = req.user;
   res.locals.path = req.path;
   res.locals.env  = app.get('env');
+  res.locals._ = require('underscore');
```
Comment (MrMaksimize) [inline app.js]: Sorry -- should be require lodash.

I've been trying to factor underscore out gradually.  We have it all over the place -- sometimes lodash, sometimes underscore.  Should just use one, and lodash is more complete.

But AWESOME freaking idea putting it into the locals.  I wonder if there's a performance implication though

File: models/defs/Opportunity.js [hunk 4]
```diff
@@ -194,11 +194,13 @@ module.exports = function(sequelize, DataTypes) {
 
   classMethods = _.extend(modelUtils.classMethods, {
     getListFields: function() {
-      /*return [
```
Comment (MrMaksimize) [inline models/defs/Opportunity.js]: Nice!

File: views/admin/entry.jade [hunk 5]
```diff
@@ -17,4 +17,4 @@ block content
             tr
               td
                 strong= field.label
-              td= entry.selectedValues[field.name]
+              td= entry.dataValues[field.name]
```
Comment (MrMaksimize) [inline views/admin/entry.jade]: just curious -- why did you change from selectedValues to dataValues here.  

I feel like data values object shouldn't be getting access directly, but more of a pattern following here:

http://take.ms/Neakd

**Comment by MrMaksimize** [comment]
@chrisrodz just two small things left then g2g

File: views/admin/entry.jade (hunk 5 above)
Comment (chrisrodz) [inline views/admin/entry.jade]: This could be changed. It's just the first way I found out how to access
the data. Also the change was because selectedValues was removed when we
upgraded Sequelize versions. I mentioned this yesterday, remember?

On Sat, Jul 19, 2014 at 1:08 PM, Maksim Pecherskiy <notifications@github.com

> wrote:
> 
> In views/admin/entry.jade:
> 
> > @@ -17,4 +17,4 @@ block content
> >              tr
> >                td
> >                  strong= field.label
> > -              td= entry.selectedValues[field.name]
> > -              td= entry.dataValues[field.name]
> 
> just curious -- why did you change from selectedValues to dataValues here.
> 
> I feel like data values object shouldn't be getting access directly, but
> more of a pattern following here:
> 
> http://take.ms/Neakd
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/43/files#r15145600.

## 

Best,
Christian A. Rodriguez

**Comment by chrisrodz** [comment]
At a hackathon. Will work on this tomorrow afternoon.

On Sat, Jul 19, 2014 at 1:08 PM, Maksim Pecherskiy <notifications@github.com

> wrote:
> 
> @chrisrodz https://github.com/chrisrodz just two small things left then
> g2g
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/43#issuecomment-49528417.

## 

Best,
Christian A. Rodriguez

**Comment by MrMaksimize** [comment]
KK :)

On Sat, Jul 19, 2014 at 1:11 PM, Christian A. Rodriguez <
notifications@github.com> wrote:

> At a hackathon. Will work on this tomorrow afternoon.
> 
> On Sat, Jul 19, 2014 at 1:08 PM, Maksim Pecherskiy <
> notifications@github.com
> 
> > wrote:
> > 
> > @chrisrodz https://github.com/chrisrodz just two small things left
> > then
> > g2g
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/CoquiCoders/bizwallet/pull/43#issuecomment-49528417.
> 
> ## 
> 
> Best,
> Christian A. Rodriguez
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/43#issuecomment-49528461.

## 

_Maksim Pecherskiy_
Code For America http://codeforamerica.org | _Puerto Rico (team blog
http://coquicoders.org)_
_773.677.7755 || Schedule A Meeting https://calendly.com/maksim/ *
*@MrMaksimize http://twitter.com/MrMaksimize || MrMaksimize.com
http://mrmaksimize.com_
</>

**Comment by MrMaksimize** [comment]
I've merged a ton of fixes -- probably a lot of them relevant here into https://github.com/CoquiCoders/bizwallet/pull/33

After we merge that one, go through this one and see what you still wanna take in

**Comment by MrMaksimize** [comment]
So I'll mark it hold for now

**Comment by chrisrodz** [comment]
@MrMaksimize I merged our new and shiny `master` branch into this one. The fixes here are very important. Please review and merge when you can :smile:
